src/coreclr/nativeaot/System.Private.StackTraceMetadata/src/Internal/StackTraceMetadata/MethodNameFormatter.cs

```diff
@@ -181,6 +181,39 @@ private void EmitMethodParameters(MethodSignature methodSignature)
             _outputBuilder.Append(')');
         }
 
+        /// <summary>
+        /// Emit parenthesized method argument type list with parameter names.
+        /// </summary>
+        /// <param name="methodHandle">Method handle to use for parameter formatting</param>
+        private void EmitMethodParameters(MethodHandle methodHandle)
+        {
+            Method method = methodHandle.GetMethod(_metadataReader);
+            HandleCollection typeVector = method.Signature.GetMethodSignature(_metadataReader).Parameters;
+            ParameterHandleCollection.Enumerator names = method.Parameters.GetEnumerator();
+
+            bool first = true;
+
+            _outputBuilder.Append('(');
+            foreach (Handle handle in typeVector)
+            {
+                if (first)
+                {
+                    first = false;
+                }
+                else
+                {
+                    _outputBuilder.Append(", ");
+                }
+
+                names.MoveNext();
```

Comment:
I assume MethodSignature.Parameters and Method.Parameters will always have the same length?!

Reply:
They won't - `Parameters` is allowed to have (arbitrary) holes in it - we should look at the `Sequence` of the parameter to make sure it matches.

A good test case would be to have a method that has a custom attribute on the return type:

```
class AttAttribute : Attribute { }

static class Program
{
    [return: Att]
    static void Main(string[] args)
    {
    }
}
```

There will be an extra `Parameter` record with an empty name for the return value that doesn't normally show up.

Reply:
I looked around a bit to understand this better and believe what you wrote corresponds ECMA 335 II.22.33.

> Sequence shall have a value >= 0 and <= number of parameters in owner method. A
Sequence value of 0 refers to the owner method’s return type; its parameters are then
numbered from 1 onwards 
> Successive rows of the Param table that are owned by the same method shall be
ordered by increasing Sequence value - although gaps in the sequence are allowed

If i understand that correctly, the expected behaviour is like this:
```
typeVector := [int, string, object, bool]
parameters := [{seq=0}, {seq=1, name=a}, {seq=3, name=c}, {seq=4, name=null}]
=> (int a, string, object c, bool)
```


If that's not correct I need some additional pointer, I'm afraid.

Reply:
That's correct!

Reply:
Fixed it. The code got a bit complicated though.